server/service/hosts.go

```diff
@@ -63,5 +68,25 @@ func (svc Service) DeleteHosts(ctx context.Context, ids []uint, opts fleet.HostL
 	if len(hostIDs) == 0 {
 		return nil
 	}
+
+	err = svc.checkWriteForHostIDs(ctx, hostIDs)
+	if err != nil {
+		return err
+	}
 	return svc.ds.DeleteHosts(ctx, hostIDs)
 }
+
+func (svc Service) checkWriteForHostIDs(ctx context.Context, ids []uint) error {
+	for _, id := range ids {
+		host, err := svc.ds.Host(ctx, id)
```

Comment:
This could be quite slow if we try to delete a lot of hosts at once, ideally we'd have some kind of batch-loading so we don't end up running N queries (it could even filter just those that can be deleted and return the allowed list of ids). But I understand that for now we have an issue to fix, so optimization can come later.